internal/pkg/cli/project_delete.go

```diff
@@ -0,0 +1,251 @@
+	termprogress "github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/progress"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/prompt"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+const (
+	defaultProfile = "default"
+
+	fmtConfirmProjectDeletePrompt = "Are you sure you want to delete project %s?"
+	confirmProjectDeleteHelp      = "Deleting a project will remove all associated resources. (apps, envs, etc.)"
+)
+
+var (
+	errOperationCancelled = errors.New("operation cancelled")
+)
+
+// BuildProjectDeleteCommand builds the `project delete` subcommand.
+func BuildProjectDeleteCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Delete all resources associated with the local project.",
+		Example: `
+  /code $ ecs-preview project delete --yes`,
+		RunE: runCmdE(func(cmd *cobra.Command, args []string) error {
+			opts, err := newDeleteProjOpts()
+			if err != nil {
+				return err
+			}
+
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+
+			if err := opts.Ask(); err != nil {
+				return err
+			}
+
+			return opts.Execute()
+		}),
+	}
+
+	cmd.Flags().Bool(yesFlag, false, yesFlagDescription)
+	viper.BindPFlag(yesFlag, cmd.Flags().Lookup(yesFlag))
+	cmd.Flags().String(profileFlag, defaultProfile, profileFlagDescription)
+	viper.BindPFlag(profileFlag, cmd.Flags().Lookup(profileFlag))
+
+	return cmd
+}
+
+type deleteProjOpts struct {
+	*GlobalOpts
+
+	projectName      string
+	skipConfirmation bool
+	profile          string
+
+	sessProvider sessionProvider
+	store        projectService
+	deployer     deployer
+	ws           workspaceDeleter
+	spinner      progress
+}
+
+type workspaceDeleter interface {
+	Delete() error
+}
+
+func newDeleteProjOpts() (*deleteProjOpts, error) {
+	store, err := store.New()
+	if err != nil {
+		return nil, err
+	}
+
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, err
+	}
+
+	s, err := session.NewProvider().Default()
+	if err != nil {
+		return nil, err
+	}
+	cf := cloudformation.New(s)
+
+	return &deleteProjOpts{
+		GlobalOpts:       NewGlobalOpts(),
+		projectName:      viper.GetString(projectFlag),
+		skipConfirmation: viper.GetBool(yesFlag),
+		profile:          viper.GetString(profileFlag),
+		store:            store,
+		ws:               ws,
+		deployer:         cf,
+		spinner:          termprogress.NewSpinner(),
+	}, nil
+}
+
+func (o deleteProjOpts) Validate() error {
+	if o.ProjectName() == "" {
+		return errNoProjectInWorkspace
+	}
+
+	return nil
+}
+
+func (o deleteProjOpts) Ask() error {
+	if o.skipConfirmation {
+		return nil
+	}
+
+	manualConfirm, err := o.prompt.Confirm(
+		fmt.Sprintf(fmtConfirmProjectDeletePrompt, o.ProjectName()),
+		confirmProjectDeleteHelp,
+		prompt.WithTrueDefault())
+
+	if err != nil {
+		return err
+	}
+
+	if !manualConfirm {
+		return errOperationCancelled
+	}
+
+	return nil
+}
+
+func (o deleteProjOpts) Execute() error {
```

Comment:
Since we have `deleteApps` I think it would be easier to read if we also have `deleteProject`, `deleteEnvs` and so on.

Reply:
`deleteProject` and `deleteEnvs` was a bit trickier since I need to share the accounts and regions from the environments for project stackset deletion so I opted to keep it inline for now to prevent the need to store more stuff unnecessarily in the opts structure.